R4R: Minting mechanism explanatory docs (#3894)
cosmos/cosmos-sdk: docs/spec/mint/01_concepts.md

# Concepts

## The Minting Mechanism

The minting mechanism was designed to:

>  - allow for a market evaluation of the inflation rate

Reviewer: ```suggestion
 - allow for a flexible inflation rate determined by market demand targeting a particular bonded-stake ratio
```

>  - optimize between market liquidity and staked supply

Reviewer: ```suggestion
 - effect a balance between market liquidity and staked supply
```

>  - optimize between market liquidity and staked supply.

Reviewer 2: ```suggestion
 - optimize between market liquidity and staked supply
```

In order to best determine the appropriate market rate for inflation rewards, a
moving change rate is used.  The moving change rate mechanism ensures that if

> the % bonded is either over of under the goal %-bonded, the inflation rate will

Reviewer: ```suggestion
the % bonded is either over of or under the goal %-bonded, the inflation rate will
```

> adjust to further incentivize or incentive being bonded. Setting the goal

Reviewer 2: `to further incentivize or incentive being bonded.` ...seems a bit off to me, no?

Author: whoops .. updated `incentivize or incentive` -> `incentivize or disincentivize`

> adjust to further incentivize or disincentivize being bonded. Setting the goal

Reviewer: ```suggestion
adjust to further incentivize or disincentivize being bonded, respectively. Setting the goal
```

%-bonded at less than 100% encourages the network to maintain some non-staked tokens

> which presumably encourages liquidity. 

Reviewer: ```suggestion
which should help provide some liquidity.
```

It can be broken down in the following way: 
 - If the inflation rate is below the goal %-bonded the inflation rate will

>    increase until a maximum cap is reached

Reviewer: ```suggestion
   increase until a maximum value is reached
```

 - If the goal % bonded (67% in Cosmos-Hub) is maintained, then the inflation
   rate will stay constant 
 - If the inflation rate is above the goal %-bonded the inflation rate will

>    decrease until a minimum cap is reached

Reviewer: ```suggestion
   decrease until a minimum value is reached
```
